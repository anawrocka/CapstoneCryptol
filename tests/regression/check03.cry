nats : [32] -> [inf][32]
nats n = [n] # nats (n + 1)

check3 = (nats 0 @ 45 == 45) && (nats 45 @ 0 == 45)
