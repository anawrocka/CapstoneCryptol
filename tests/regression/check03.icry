:load check03.cry
nats 0 @@ [ 0 .. 10 ]
nats 11 @@ [ 0 .. 10 ]
check3
