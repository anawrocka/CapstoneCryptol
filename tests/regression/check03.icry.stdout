Loading module Cryptol
Loading module Cryptol
Loading module Main
[warning] at <interactive>:1:1--1:22:
  Defaulting type parameter 'bits'
             of finite enumeration
             at <interactive>:1:11--1:22
  to 4
[0x00000000, 0x00000001, 0x00000002, 0x00000003, 0x00000004,
 0x00000005, 0x00000006, 0x00000007, 0x00000008, 0x00000009,
 0x0000000a]
[warning] at <interactive>:1:1--1:23:
  Defaulting type parameter 'bits'
             of finite enumeration
             at <interactive>:1:12--1:23
  to 4
[0x0000000b, 0x0000000c, 0x0000000d, 0x0000000e, 0x0000000f,
 0x00000010, 0x00000011, 0x00000012, 0x00000013, 0x00000014,
 0x00000015]
True
